mlflow/transformers/llm_inference_utils.py

```diff
@@ -8,13 +22,133 @@
 _LLM_INFERENCE_TASK_COMPLETIONS = "llm/v1/completions"
 _LLM_INFERENCE_TASK_CHAT = "llm/v1/chat"
 
+_LLM_INFERENCE_OBJECT_NAME = {
+    _LLM_INFERENCE_TASK_COMPLETIONS: "text_completion",
+    _LLM_INFERENCE_TASK_CHAT: "chat.completion",
+}
+
 _SUPPORTED_LLM_INFERENCE_TASK_TYPES_BY_PIPELINE_TASK = {
     "text-generation": [_LLM_INFERENCE_TASK_COMPLETIONS, _LLM_INFERENCE_TASK_CHAT],
 }
 
+_SIGNATURE_FOR_LLM_INFERENCE_TASK = {
+    _LLM_INFERENCE_TASK_CHAT: ModelSignature(
+        inputs=CHAT_MODEL_INPUT_SCHEMA, outputs=CHAT_MODEL_OUTPUT_SCHEMA
+    ),
+    _LLM_INFERENCE_TASK_COMPLETIONS: ModelSignature(
+        inputs=COMPLETIONS_MODEL_INPUT_SCHEMA, outputs=COMPLETIONS_MODEL_OUTPUT_SCHEMA
+    ),
+}
+
+
+def infer_signature_from_llm_inference_task(
+    inference_task: str, signature: Optional[ModelSignature] = None
+) -> ModelSignature:
+    """
+    Infers the signature according to the MLflow inference task.
+    Raises exception if a signature is given.
+    """
+    if signature is not None:
+        raise MlflowException(
+            f"When `task` is specified as `{inference_task}`, the signature would "
+            "be set by MLflow. Please do not set the signature."
+        )
+
+    return _SIGNATURE_FOR_LLM_INFERENCE_TASK[inference_task]
+
+
+def convert_data_messages_with_chat_template(data, tokenizer):
+    """For the Chat inference task, apply chat template to messages to create prompt."""
+    try:
+        messages = data.pop("messages").tolist()[0]
+    except KeyError:
+        raise MlflowException("The 'messages' field is required for the Chat inference task.")
+
+    try:
+        messages_str = tokenizer.apply_chat_template(
+            messages, tokenize=False, add_generation_prompt=True
+        )
+    except Exception as e:
+        raise MlflowException(f"Failed to apply chat template: {e}")
+
+    data["prompt"] = messages_str
+
+
+def preprocess_llm_inference_params(
+    data,
+    flavor_config: Optional[Dict[str, Any]] = None,
+) -> Tuple[List[Any], Dict[str, Any]]:
+    """
+    When a MLflow inference task is given, return updated `data` and `params` that
+    - Extract the parameters from the input data.
+    - Replace OpenAI specific parameters with Hugging Face specific parameters, in particular
+      - `max_tokens` with `max_new_tokens`
+      - `stop` with `stopping_criteria`
+    """
+    updated_data = []
+    params = {}
+
+    for column in data.columns:
+        if column in ["prompt", "messages"]:
+            updated_data = data[column].tolist()
```

Comment:
Is data always a pandas DataFrame here?

Reply:
It should be a pandas DataFrame after current _enforce_schema check, but we'd better add a type check in case the logic changes in the future and then the error message would be confusing :)

Reply:
Got it, I added a type check and error message above now.